Refactoring after #1481 - removing not needed env variables

pszkamruk-splunk opened the pull request:
**Description:** 
Cleanup and refactor work to prevent code duplication related to setting Kubernetes attributes by using k8sAttributesProcessor for cluster receiver introduced by [PR](https://github.com/signalfx/splunk-otel-collector-chart/pull/1481)

Once following two PRs will be merged we can merge this one:
 - [Contrib repo PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35528)
 - [Chart repo PR](https://github.com/signalfx/splunk-otel-collector-chart/pull/1481)
 
 We need also decide what to do with `resource/add_collector_k8s`: [link](https://github.com/signalfx/splunk-otel-collector-chart/blob/7ea0cab65fd0825b61e162e405ae11bedcf06890/helm-charts/splunk-otel-collector/templates/config/_otel-k8s-cluster-receiver-config.tpl#L121)
as this is using env variables which we want to remove, but the functionality of adding additional parameters is covered by `k8sattributes/metrics`, so probably it is safe to remove `resource/add_collector_k8s` from configuration

jvoravong commented:
For resource/add_collector_k8s:
  - Replace this with a k8sattributes processor.
  - Aim to use a single instance of the k8sattributes processor if possible. If required, add a second instance (e.g., k8sattributes/add_collector) for handling collector metrics specifically.
  - Functional Tests cover internal collector metrics and attributes (e.g., [otelcol_exporter_send_failed_spans](https://github.com/signalfx/splunk-otel-collector-chart/blob/0e083695672ef3055a008964f6c33675dea236f4/functional_tests/testdata/expected_kind_values/expected_internal_metrics.yaml#L5044)). Check to see if you can get the tests to pass with using only a single k8sattributes processor instance. Add a second processor instance (named something like `k8sattributes/add_collector`) because the tests failed and it is necessary.

atoulme commented:
sorry, please fix conflicts

pszkamruk-splunk commented:
@jvoravong 

Regarding work done here:
1. Introduced logs and metrics attributes tests as a separate PR - #1610
2. Removed environment variables from [deployment-cluster-receiver.yaml](https://github.com/signalfx/splunk-otel-collector-chart/pull/1540/files#diff-e48639a1114468a898dae03f941f515324d03bb284ef976da41ffae2726b831e)
K8S_NODE_NAME - is still there because this field is used by `k8sattributes/add_collector` to add `k8s.node.name` which is later used by`splunk-otel-collector.k8sAttributesSplunkPlatformMetrics` for matching, after which additional attributes are added by `k8sAttributesProcessor`
3. We need to keep `initContainers.env` variables ([link](https://github.com/signalfx/splunk-otel-collector-chart/blob/main/helm-charts/splunk-otel-collector/templates/deployment-cluster-receiver.yaml#L94)) - these are used by `/splunk-scripts/init-eks-fargate-cluster-receiver.sh` 
4.  updated `metadata` section at `splunk-otel-collector.k8sAttributesSplunkPlatformMetrics` - previously no attributes were added `[]` - this also caused that functional test `kubernetes_cluster_metrics` had to be updated 

Once this PR will be merged, next steps are: 
1. Unify `splunk-otel-collector.k8sClusterReceiverAttributesProcessor`, `splunk-otel-collector.k8sAttributesProcessor`, `splunk-otel-collector.k8sAttributesSplunkPlatformMetrics` into one `k8sAttributesProcessor`
2. A change for functional tests will be needed
3. I am also considering adding `k8sattributes/add_collector` and then using it for agent and clusterReceiver pipelines in similar way as it is used right now for metrics in clusterReceiver

PS: Let's merge first #1610 as I pulled it into this PR, then I will rebase and squash and we can merge this one.

dmitryax commented:
The `k8sattributes` processor requires the pulling the k8s API state from the whole cluster into the collector cache, which can take a lot of memory. Currently, if logs pipeline isn't enabled, the k8sattributes processor isn't required. We should not introduce this overhead this just to reduce code.

pszkamruk-splunk commented:
We are not going to merge this. Closing this PR
